2014-04-24  Magnus Granberg  <zorry@gentoo.org>

		* gcc/doc/invoke.texi	Add NOTES about -fstack-protector-all, -pie and
		-fPIE/-fpie when --enable-esp is enable, this options is on by default.

--- a/gcc/doc/invoke.texi	2009-04-01 09:18:47.000000000 +0200
+++ b/gcc/doc/invoke.texi	2009-06-18 14:08:38.000000000 +0200
@@ -9233,6 +9245,11 @@ If a guard check fails, an error message
 @opindex fstack-protector-all
 Like @option{-fstack-protector} except that all functions are protected.
 
+NOTE: NOTE: When --enable-esp this option is enabled by default 
+for C, C++, ObjC, ObjC++, if neither @option{-fno-stack-protector}, 
+@option{-nostdlib}, @option{-ffreestanding}, @option{-fstack-protector}, 
+@option{-fstack-protector-strong}or @option{-fstack-protector-all}are found.
+
 @item -fstack-protector-strong
 @opindex fstack-protector-strong
 Like @option{-fstack-protector} but includes additional functions to
@@ -7960,6 +7965,12 @@
 that were used to generate code (@option{-fpie}, @option{-fPIE},
 or model suboptions) when you specify this option.
 
+NOTE: When --enable-esp this option is enabled by default
+for C, C++, ObjC, ObjC++, if neither @option{-fno-pie} or @option{-fno-PIE}
+or @option{-fno-pic} or @option{-fno-PIC} or @option{-nostdlib} or
+@option{-nostartfiles} or @option{-shared} or @option{-pg} or @option{-p}
+are found.
+
 @item -rdynamic
 @opindex rdynamic
 Pass the flag @option{-export-dynamic} to the ELF linker, on targets
@@ -15889,6 +15910,11 @@
 @code{__pie__} and @code{__PIE__}.  The macros have the value 1
 for @option{-fpie} and 2 for @option{-fPIE}.
 
+NOTE: When --enable-esp this option is enabled by default
+for C, C++, ObjC, ObjC++, if neither @option{-fno-pie} or @option{-fno-PIE}
+or @option{-fno-pic} or @option{-fno-PIC} or @option{-nostdlib} or
+@option{-nostartfiles} or @option{-shared} are found.
+
 @item -fno-jump-tables
 @opindex fno-jump-tables
 Do not use jump tables for switch statements even where it would be
